AnnoyancesFilter/sections/cookies_specific.txt

```diff
@@ -5358,6 +5358,7 @@ selgros.pl#%#//scriptlet('set-cookie', 'cookie_new_msg', '1')
 guard.io#%#//scriptlet('set-cookie', 'cookie_consent_seen', '1')
 consent.youtube.com#%#AG_onLoad(function(){if(window.location.href.includes("consent.youtube.com")){var b=new MutationObserver(function(){var a=document.querySelector('form[action="https://consent.youtube.com/s"] > input[type="submit"]');a&&a.click();console.log('sadsa');});b.observe(document,{childList:!0,subtree:!0});setTimeout(function(){b.disconnect()},1E4)}});
 meyer-gartenbedarf.de#%#//scriptlet('set-cookie', 'amcookie_allowed', '1')
+tennisassa.fi#%#AG_onLoad(function() { setTimeout(function() { var el = document.querySelector('form[action="/cookiewall/Save"] > .js-simple-accept-view > button[type="submit"]'); if(el) { el.click(); } }, 300); });
```

Comment:
Better use `MutationObserver` with a conditional.
```
tennisassa.fi#%#AG_onLoad(function(){if(location.pathname === '/cookiewall/'){var a=new MutationObserver(function(){var b=document.querySelector(".js-accept-basic");b&&(a.disconnect(),b.click())});a.observe(document,{childList:!0,subtree:!0});setTimeout(function(){a.disconnect()},1E4)}});
```

Reply:
I don't think `MutationObserver` is needed in this case.

Reply:
@zloyden I have also tried a similar rule before adding a simpler rule. This rule doesn't click the Accept Cookies at first visit, it works fine after refreshing the site.

Reply:
> I don't think `MutationObserver` is needed in this case.

Why so? Your rule will apply to the entire part of the website, you also depend on a delay of `300`.


> I have also tried a similar rule before adding a simpler rule. This rule doesn't click the Accept Cookies at first visit, it works fine after refreshing the site.

Try without `AG_onLoad`

```
tennisassa.fi#%#(function(){if("/cookiewall/"===location.pathname){var a=new MutationObserver(function(){var b=document.querySelector(".js-accept-basic");b&&(a.disconnect(),b.click())});a.observe(document,{childList:!0,subtree:!0});setTimeout(function(){a.disconnect()},1E4)}})();
```

Reply:
It works now. Btw, why didn't it work with `AG_onLoad`?

Reply:
When it's used with `AG_onLoad`, probably because there is not changes in DOM.